### DISCUSSION TITLE
How to deal with the error serializing dates when they're returned from getServerSideProps?

### DISCUSSION THREAD

**Discussion by lumenwrites** [vercel/next.js#44847] [Help]
Whenever I try to return a date object from getServerSideProps, I get an error:
```
Error: Error serializing `.post.createdAt` returned from `getServerSideProps` in "/community/[postSlug]".
```
I don't want to manually loop through all the posts on every request and delete (or stringify) the date field.

Is there some automatic way to fix this issue all throughout my next.js project?

**Comment by icyJoseph** [comment]
People often do a little hack:

```tsx

return {
   props: JSON.parse(JSON.stringify(data))
}
```

That'll take care of it, albeit, it is not very nice. People who use MongoDB use that hack a lot.

For example: https://www.mongodb.com/developer/languages/javascript/nextjs-with-mongodb/#example-2--next-js-pages-with-mongodb (look through the snippet under this heading)

**Comment by GabenGar** [comment]
Do not use `Date` object for serialization. Instead use RFC 3339 date string to interact between frontend and backend. This will also prevent date-related hydration mismatch errors and eventual redux store errors.
